### PR TITLE
[Fix] fix train example

### PR DESCRIPTION
**Pull request by teamwong111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Related issue: https://github.com/open-mmlab/mmcv/issues/1470.
Some users want to test the example/train.py on single GPU even though there are multiple GPUs on their machines.
Force to use GPU:0 can avoid Error traceback.

## Modification

Assign `device_ids = [0]` and fix a format error by the way.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zhouzaida** [comment]
Please add the following warning information in the docstring.

[mmcv/parallel/data_parallel.py](https://github.com/open-mmlab/mmcv/blob/add157cc73b05c7be3d9b63505e17dc2288a6d0f/mmcv/parallel/data_parallel.py#L18)
```python
.. warning::
    MMDataParallel only supports single GPU training, if you need to train with multiple GPUs, please use MMDistributedDataParallel instead. If you have multiple GPUs and you just want to use MMDataParallel, you can set the ``CUDA_VISIBLE_DEVICES=0`` or ``MMDataParallel(model, device_ids=[0])``.
```

File: examples/train.py [hunk 1]
```diff
@@ -42,7 +42,9 @@ def train_step(self, data, optimizer):
 if __name__ == '__main__':
     model = Model()
     if torch.cuda.is_available():
-        model = MMDataParallel(model.cuda())
+        # only use gpu:0 to train
+        # Solved issue https://github.com/open-mmlab/mmcv/issues/1470
+        model = MMDataParallel(model.cuda(), [0])
```
Comment (zhouzaida) [inline examples/train.py:47]:
```suggestion
        model = MMDataParallel(model.cuda(), device_ids=[0])
```

**Comment by teamwong111** [comment]
> Please add the following warning information in the docstring.
> 
> [mmcv/parallel/data_parallel.py](https://github.com/open-mmlab/mmcv/blob/add157cc73b05c7be3d9b63505e17dc2288a6d0f/mmcv/parallel/data_parallel.py#L18)
> 
> ```python
> .. warning::
>     MMDataParallel only supports single GPU training, if you need to train with multiple GPUs, please use MMDistributedDataParallel instead. If you have multiple GPUs and you just want to use MMDataParallel, you can set the ``CUDA_VISIBLE_DEVICES=0`` or ``MMDataParallel(model, device_ids=[0])``.
> ```

Actually, set the ``CUDA_VISIBLE_DEVICES=0`` is useless.

File: mmcv/parallel/data_parallel.py [hunk 2]
```diff
@@ -15,6 +15,13 @@ class MMDataParallel(DataParallel):
       flexible control of input data during both GPU and CPU inference.
     - It implement two more APIs ``train_step()`` and ``val_step()``.
 
+    .. warning::
+        MMDataParallel only supports single GPU training, if you need to
+        train with multiple GPUs, please use MMDistributedDataParallel
+        instead. If you have multiple GPUs and you just want to use
+        MMDataParallel, you can set the ``CUDA_VISIBLE_DEVICES=0`` or
+        ``MMDataParallel(model, device_ids=[0])``.
```
Comment (zhouzaida) [inline mmcv/parallel/data_parallel.py:23]:
```suggestion
        MMDataParallel, you can set the environment variable  ``CUDA_VISIBLE_DEVICES=0`` or
        instantiate ``MMDataParallel`` with ``device_ids=[0]``.
```

**Review by zhouzaida — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**